rules.py

```diff
@@ -941,43 +941,41 @@ def check(self, element):
                 element.sourceline, element.get("objectId")))
 
 class AllCaps(base.BaseRule):
-    """The Name elements in Candidates, Contests and Person elements should not be in all uppercase.
-
-    If the name elements in Candidates, Contests and Person elements are in uppercase, 
-    the list of objectIds of those elements will be returned to the user as a warning."""
+    """The Name elements in Candidates, Contests and Person elements should not
+    be in all uppercase.
 
+    If the name elements in Candidates, Contests and Person elements are in
+    uppercase, the list of objectIds of those elements will be returned to
+    the user as a warning.
+    """
+    
     def elements(self):
         return ["Candidate", "CandidateContest", "Person"]
-
+    
     def check(self, element):
         object_id = element.get("objectId")
         if element.tag == "Candidate":
-            ballot_name = element.find("BallotName")
+            ballot_name = element.find("BallotName").find("Text")
             if ballot_name is not None:
-                if ballot_name.find("Text") is not None:
-                    name = ballot_name.find("Text").text
-                    if name is not None and name == name.upper():
-                        raise base.ElectionWarning(
-                            "Line %d. Candidate %s has name in all upper case letters." % (
-                            element.sourceline, object_id))
-        elif element.tag == "Contest":
-            name_element = element.find("Name")
-            if name_element is not None:
-                name = name_element.text
-                if name is not None and name == name.upper():
+                name = ballot_name.text
+                if name and name.isupper():
                     raise base.ElectionWarning(
-                        "Line %d. Contest %s has name in all upper case letters." % (
-                        element.sourceline, object_id))
+                        "Line %d. Candidate %s has name in all upper case letters." %
```

Comment:
Please correct typo in this case and use Contest Instead.

Reply:
Actually this particular warning message is for the Candidate, so I included 'Candidate' in the message. The one for 'Contest' is in the next conditional block.

Reply:
Looks like the code sections before and after this change were misaligned, and you are correct. Thanks for the clarification!